Frameworks/Foundation/NSIndexPath.mm

```diff
@@ -90,114 +104,90 @@ - (BOOL)isEqual:(id)other {
 /**
  @Status Interoperable
 */
-- (void)dealloc {
-    if (_indexes) {
-        IwFree(_indexes);
-    }
-    [super dealloc];
-}
-
-/**
- @Status Caveat
- @Notes position must be valid
-*/
-- (unsigned)indexAtPosition:(unsigned)position {
-    if (position >= _length) {
-        assert(0);
-        //[NSException raise:NSInvalidArgumentException format:@"Unable to remove index from zero length path."];
-        return 0;
-    }
-
-    return _indexes[position];
+- (NSUInteger)indexAtPosition:(NSUInteger)position {
+    return position < _indexes.size() ? _indexes[position] : NSNotFound;
 }
 
 /**
  @Status Interoperable
 */
 - (void)getIndexes:(NSUInteger*)indexes {
-    for (unsigned int i = 0; i < _length; i++) {
-        indexes[i] = _indexes[i];
-    }
+    std::copy(_indexes.cbegin(), _indexes.cend(), indexes);
 }
 
 /**
  @Status Interoperable
 */
 - (id)copyWithZone:(NSZone*)zone {
-    return [[[self class] alloc] initWithIndexes:_indexes length:_length];
+    return [[[self class] alloc] initWithIndexes:_indexes.data() length:_indexes.size()];
 }
 
 /**
  @Status Interoperable
 */
 - (NSIndexPath*)indexPathByAddingIndex:(NSUInteger)newIndex {
-    id ret = [[self class] alloc];
-
-    NSUInteger* indexCopy = (NSUInteger*)IwMalloc((_length + 1) * sizeof(DWORD));
-    memcpy(indexCopy, _indexes, _length * sizeof(DWORD));
-    indexCopy[_length] = newIndex;
-
-    [ret initWithIndexes:indexCopy length:_length + 1];
-
-    IwFree(indexCopy);
+    std::vector<NSUInteger> indexes = _indexes;
+    indexes.emplace_back(newIndex);
+    return [NSIndexPath indexPathWithIndexes:indexes.data() length:indexes.size()];
+}
 
-    return [ret autorelease];
+/**
+ @Status Interoperable
+*/
+- (NSIndexPath*)indexPathByRemovingLastIndex {
+    return [NSIndexPath indexPathWithIndexes:_indexes.data() length:(_indexes.empty() ? 0 : _indexes.size() - 1)];
 }
 
 /**
  @Status Interoperable
 */
 - (NSUInteger)length {
-    return _length;
+    return _indexes.size();
 }
 
 /**
  @Status Interoperable
 */
 - (NSUInteger)hash {
-    unsigned ret = 0;
-
-    for (unsigned int i = 0; i < _length; i++) {
-        ret += _indexes[i];
-    }
-
-    return ret;
+    // Allegedly the behavior of the reference platform, does not seem to be so for test machines but it will suffice
```

Comment:
I was told this implementation was written this way because of observed values, but the reference platform seems to have updated the hash implementation.  I'll change this to something more suitable as well.  It won't be equivalent to reference platform, but I don't think it needs to be